### PR TITLE
Adding context Contact Preference element to Join Form

### PR DESCRIPTION
**Pull request by had2020**

### CODE REVIEW

**Comment by had2020** [comment]
this adds the element needed for the Add "Contact Preference field to member object" issue

File: components/JoinForm/JoinForm.tsx [hunk 1]
```diff
@@ -117,6 +117,8 @@ const JoinForm = () => {
             defaultCountry="US"
             value={phoneNumber}
             onChange={setPhoneNumber}/>
+            
+          <input type="text" name="ContactPreference" placeholder="Contact Preference" required />
```
Comment (WillNilges) [inline components/JoinForm/JoinForm.tsx:121]: This should be a set of radio buttons with two options:
- `Email`
- `Phone`

Email should be the default.

**Review by WillNilges — CHANGES_REQUESTED** [review]
:wave: Thanks for the PR. Is this supposed to be a draft? If so, cool, but if not, there's more that needs doing here.

- I see that you have a PR open in `meshdb` (https://github.com/nycmeshnet/meshdb/pull/452). Please get those changes merged first, so that the field has somewhere to go.
    - While you're at it, update the Join Form endpoint to accept this new data, but make sure that it's optional (so that the join form will still work until this PR gets merged)
- Update the Meshforms API model to send the new data
- Write some unit tests to make sure that the Join Form is working properly. We have playwright setup in this repo, so you could use that and create some kind of mock endpoint.

**Comment by WillNilges** [comment]
Also can you link the ticket from our backlog/launch tracker? 
https://github.com/orgs/nycmeshnet/projects/4/views/1

**Pull request closed without merging**